### ISSUE TITLE
Add an option for one-based column indexes

### ISSUE THREAD

**Issue by jeremija** [eslint/eslint#2261]
Currently only zero-based column indexes are used. I would like to have a `-i` / `--one-based-index` flag which enables one-based column indexes.

The reason I'd like to have this functionality is because Sublime Text's default build system uses one-based indexes and thus all errors selected with `F4` are off by one character. Here is the example of a `.sublime-build` config:

```json
{
    "cmd": ["eslint", "$file"],
    "file_regex": "^(.+\\.js)$",
    "line_regex": " *([0-9]+):([0-9]+) *(.+)$",
    "selector": ["source.js"]
}
```

For example:

```bash
~/src/js/eslint$ bin/eslint.js tests/fixtures/missing-semicolon.js 

tests/fixtures/missing-semicolon.js
  1:4   error  foo is defined but never used  no-unused-vars
  1:15  error  Missing semicolon              semi

✖ 2 problems (2 errors, 0 warnings)
```

And with the `--one-based-index` flag:

```bash
~/src/js/eslint$ bin/eslint.js --one-based-index tests/fixtures/missing-semicolon.js 

tests/fixtures/missing-semicolon.js
  1:5   error  foo is defined but never used  no-unused-vars
  1:16  error  Missing semicolon              semi

✖ 2 problems (2 errors, 0 warnings)
```

I am aware of the [SublimeLinter-eslint](https://github.com/roadhump/SublimeLinter-eslint) package, but I would prefer to use something simpler like [sublime-jshint](https://github.com/uipoet/sublime-jshint). Hopefully this would also help others if they encounter a similar problem.

I already have this implemented and will create a pull request shortly.

Thanks

**Comment by nzakas** [comment]
Before sending another pull request, let's determine if it makes sense to add such a feature, as it's a bit strange. @btmills @lo1tuma do you have opinions on this?

There's also this plugin that allows you to update the column number: https://github.com/albertosantini/sublimetext-buildnext

**Comment by jeremija** [comment]
Hi - I've had success with the Build Next plugin, but it took me a while to realize it doesn't support `line_regex`, so I had to use the `--format compact` option. In case anybody needs it, here is the `file_regex` value I've used `^(.+): line ([0-9]+), col ([0-9]+)(.+)$`.

I would like to point out that a lot of other editors have one-based column indexes by default. For example Eclipse, IntelliJ IDEA, Notepad++, Gedit and Vim. Maybe they can all be reconfigured, but I think it would be nice to have an alternative.

**Comment by lo1tuma** [comment]
Couldn’t this be achieved with a custom formatter?

I’m not sure if this problem is common enough to support it. AFAICT the default eslint integration for IntelliJ IDEA shows eslint errors on the correct column.

**Comment by jeremija** [comment]
It definitely could. Is it possible to use a custom formatter which is not included in the `eslint` package? If not, would it be fine if I implemented one and submitted it?

**Comment by lo1tuma** [comment]
@jeremija Yes, this is possible `eslint -f ./path/to/customformat.js`.

**Comment by jeremija** [comment]
@lo1tuma Good to know, I didn't think this was possible. I guess there is no need for the changes I made then - thank you for your time!

**Comment by nzakas** [comment]
One thing worth bringing up in this discussion: is using zero-based columns in ESLint a bug or a feature? Since we have 1.0.0 coming up, this would be the right time to make a change if we think it's a bug.

As it is, we are just passing through what the parser gives us, but a case can be made that this is unexpected behavior. Thoughts?

**Comment by doberkofler** [comment]
@nzakas I personally see this as a bug and would generally prefer 1-based column indexes to have eslint in sync with jslint and jshint.
